ludwig/models/llm.py

```diff
@@ -152,15 +172,16 @@ def __init__(
                 # because the model has additional "head" layers that are used to predict the next
                 # token in the sequence. These head layers can add additional dimensions to the
                 # logits tensor, beyond the vocab_size dimension.
-                input_size=self.input_shape[-1] if self.output_feature_type == TEXT else self.model.config.vocab_size,
+                input_size=self.input_shape[-1] if self.output_feature_type == TEXT else self.model_config.vocab_size,
             )
         )
 
         # Extract the decoder object for the forward pass
         self._output_feature_decoder = ModuleWrapper(self.output_features.items()[0][1])
 
-        # Initialize the PEFT adapter is one is provided
-        self.initialize_adapter()
+        # Defer adapter initialization until after serialization for Ray backend
+        if not self.is_ray_backend:
+            self.initialize_adapter()
```

Comment:
Why do we defer initialization to post serialization when using an adapter?

Reply:
I don't think we need to, will revert. I initially thought we'd end up with a model structure that is PEFT(LLM()) where the LLM parameters are None but PEFT parameters actually have the weight matrixes, and during replace_tensor we'd run into an issue where we're loading the wrong weights for the wrong layers, but I don't think that will be the case.

Reply:
Actually this may still be true